### ISSUE TITLE
Not working after upgrading to El Capitan OSX 10.11

### ISSUE THREAD

**Issue by davidetruffo** [bobthecow/git-flow-completion#43]
I just upgraded my OSX and the autocomplete just doesn't work anymore (i hit tab, and i get a list of files in that folder, but not the git-flow commands)

any hint?

**Comment by bobthecow** [comment]
Not without more info :)

What shell are you using? How did you install git-flow? Do you use an os package manager or shell plugin manager? Does regular git completion work for you? Does any other tab completion (besides files) work?

**Comment by davidetruffo** [comment]
fair enough :)

i use iTerm + zsh + oh-my, and brew to install most of the packages (used to install both git and git-flow)
Regular git completion works, as well as other tab completion (besides files) such as grunt, bower, pip, pyenv, brew, and others.

let me know if you need any further infos

thanks

**Comment by joshuakarjala** [comment]
Also doesn't work here.

zsh 5.1.1 (x86_64-apple-darwin15.0.0)

git-flow and zsh installed by homebrew

git and many other tab-completions still work

**Comment by mikedevita** [comment]
@joshuakarjala what version of git are you using.. i have two macbook's both on el capitan.. one works the other doesn't..

## Working 
1. zsh 5.0.8
2. oh-my-zsh - latest from upgrade_oh_my_zsh
3. git 2.3.8

## Non Working
1. zsh 5.0.8
2. oh-my-zsh - latest from upgrade_oh_my_zsh
3. git 2.5.3

**Comment by joshuakarjala** [comment]
1. zsh 5.1.1
2. oh-my-zsh - latest from upgrade_oh_my_zsh
3. git 2..6.2

**Comment by mikedevita** [comment]
I think this has to do with _git completion in /usr/share .. It's not writable on capital because of the system integrity stuff. I could be wrong though. Haven't tested it yet.

**Comment by davidetruffo** [comment]
same settings as @joshuakarjala here:
- zsh 5.0.8 (x86_64-apple-darwin15.0)
- oh-my-zsh - latest from upgrade_oh_my_zsh
- git version 2.6.2
- iTerm2 Build 2.1.4
- OS X El Capitan (10.11.1)

**Comment by c0nscience** [comment]
I got the same error/issue on my fresh install mbp. I have the same environment as @davidetruffo

**Comment by istrasci** [comment]
Same settings as @davidetruffo and I also get the error.

**Comment by davidetruffo** [comment]
Hi @bobthecow,
any news on this? :)

**Comment by bobthecow** [comment]
From what I can tell, it should still be working. Looking and git and zsh source, I don't see a reason for it to break. But to be honest, I just don't know. I don't use zsh, so this version of the completion is largely user supported. For fun, can one of you replace your `_git` completion file with the current HEAD version here:

http://sourceforge.net/p/zsh/code/ci/master/tree/Completion/Unix/Command/_git?format=raw

… and see if that fixes things?

**Comment by davidetruffo** [comment]
Will try it tomorrow

**Comment by davidetruffo** [comment]
no luck.. everything seems to be up to date..
I don't have much knowledge about bash-shell-scripts, but is there any way to make it fail loudly, or to see if the plugin actually loads?

**Comment by c0nscience** [comment]
I just uninstalled git 2.6.2 and stick with the apple git version and it works all fine

**Comment by davidetruffo** [comment]
I just tried, and yes, it works properly with the apple version of git (git version 2.4.9 (Apple Git-60))

**Comment by istrasci** [comment]
I also confirm it works when downgrading to git version 2.4.9 (Apple Git-60).

**Comment by zx1986** [comment]
###                  User: zx1986
               ####                   Hostname: air
               ###                    Distro: OS X 10.11.1
       #######    #######             Kernel: Darwin
     ######################           Uptime: 2 days
    #####################             Shell: /bin/zsh
    ####################              Terminal: screen iTerm.app
    ####################              CPU: Intel Core i5-3427U CPU @ 1.80GHz
    #####################             Memory: 4 GB
     ######################           Disk: 79%
      ####################            Battery: 100
        ################              IP Address: .......
         ####     #####

zsh 5.1.1 (x86_64-apple-darwin14.5.0) with prezto
git version 2.6.3

not working :-(

**Comment by cielpy** [comment]
I have close SIP feature,it also does not work.

**Comment by cielpy** [comment]
Yes,my git installed by Homebrew, when I unlink git,use the system's git,it works.

**Comment by istrasci** [comment]
Any progress on this at all?  Still works with git version 2.5.4 (Apple Git-61), but not with the higher-versioned Homebrew ones.

**Comment by esetnik** [comment]
I faced the same issue. I haven't dug into it too much yet. It's unfortunate as Apple git is very old.

**Comment by davidetruffo** [comment]
Hi people!

Just want to highlight that:
- Mac OS 10.11.3
- zsh 5.2 (x86_64-apple-darwin15.0.0)
- git version 2.7.2 (the brewed one)
- iTerm2 2.9.20160206
using git-flow-completion as a plugin it works again

I'm not sure why though :)

**Comment by esetnik** [comment]
@davidetruffo can you provide some further configuration to help me get it going? I believe I have the same environment as you. I'm using oh-my-zsh with plugins `git-flow-avh` and `git-flow-completion`, but I've tried it with both `git-flow` and `git-flow-avh`.

**Comment by davidetruffo** [comment]
@esetnik i'll do my best :)

I am using zsh and oh-my-zsh (don't know if that makes any difference honestly)
I've also realised that i had a git-flow-completion folder in my custom plugins (in `.oh-my-zsh/custom/plugins`) but it was actually not used (i removed it, and it's still working, maybe it was a legacy from some test i've done)

In my `.zshrc` file, i have those plugins enabled (i have many others, but those are the ones related to git)
`git 
git-flow`

In the same file, i also have this line:
`fpath=(/usr/local/share/zsh-completions $fpath)`
but then again, i'm not sure if this is related.

Things that i've done recently:
- update/upgrade brew
- upgraded to iTerm2 v3 beta (the version number is 2.9.20160206)

Any better? :)

**Comment by esetnik** [comment]
@davidetruffo Thanks so much for the detailed analysis. I found two points of discrepancy between my config and yours:
1. I had `git-flow-completion` enabled as a plugin. It's actually available in the .oh-my-zsh plugins directory but you may have followed the `git-flow-completion` instructions at one point which suggests cloning this repo as a custom plugin.
2. I did not have `fpath=(/usr/local/share/zsh-completions $fpath)`

I found that a combination of removing the `git-flow-completion` from the plugins and adding `fpath=(/usr/local/share/zsh-completions $fpath)` fixed it for me as well!

I'm not sure what that magic line does but it works and it might help @bobthecow get to the bottom of this.

**Comment by davidetruffo** [comment]
@esetnik i'm glad i could help a bit :)
As far as i know, the magic line should tell zsh where to look for completions patterns. Perhaps zsh made it better and now it works. I'm not sure if this makes sense, so if anyone has some light to shed on this i'd be happy :)

**Comment by esetnik** [comment]
@davidetruffo I actually realized that the suggestions you made did not persist to a new shell. I must have had the old Apple git sourced in the old shell. I currently have everything setup the same and can't get git flow completions with the previous solution.

**Comment by davidetruffo** [comment]
@esetnik Apologies for the ridiculous late reply! Did you manage to make it work?
From my end, i can tell that it works fine now, and i removed the git-flow-completion plugin. Some details:

- Mac OS 10.11.3
- zsh 5.2 (x86_64-apple-darwin15.0.0)
- git version 2.8.2 (the brewed one)
- iTerm2 Build 2.9.20160426

**Comment by esetnik** [comment]
Unfortunately not. I can only get it working with apple git and not brewed
one.

**Comment by davidetruffo** [comment]
out of curiosity, are you using git-flow from brew as well? which version? (i have 0.4.1)

**Comment by mike3run** [comment]
Not working on osSierra either 😢

**Comment by cielpy** [comment]
use Xcode's Git will fix this problem.

**Comment by haifengkao** [comment]
confirmed. Xcode's `git version 2.7.4 (Apple Git-66)` works fine.
If you use zgen, just add `zgen oh-my-zsh plugins/git-flow` to your .zshrc
and reset zgen by `zgen reset`

**Comment by ayaz** [comment]
I am sharing my experience. I have the following with git-bash completion wasn't working:

> OS: OS X El Captian 10.11.6
> Git: git version 2.10.1 (Apple Git-78)
> Shell: GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin15)
> Homebrew: Homebrew 1.1.7

After I installed `git version 2.11.0` via brew, it _began_ working for me.

**Comment by snoepkast** [comment]
Found this issue via Google. So will post what fixed it for me:

1. `brew uninstall git`
2. `brew install git --without-completions`
3. Follow the instructions on how to use it as a plugin, doing a git clone because the one from the oh-my-zsh plugins directory didn't work for me

So now I have it working with the following versions:

* Mac OS 10.12.5
* zsh 5.3.1 (x86_64-apple-darwin16.6.0) (installed via brew)
* git version 2.13.3 (installed via brew)
